core/test/mpi/base/communicator.cpp

```diff
@@ -0,0 +1,142 @@
+are met:
+
+1. Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in the
+documentation and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#include <mpi.h>
+
+
+#include <gtest/gtest.h>
+
+
+#include <ginkgo/config.hpp>
+#include <ginkgo/core/base/mpi.hpp>
+
+
+namespace {
+
+
+class Communicator : public ::testing::Test {
+protected:
+    Communicator() : comm(MPI_COMM_WORLD) {}
+
+    void SetUp()
+    {
+        rank = comm.rank();
+        ASSERT_EQ(comm.size(), 8);
+    }
+
+    gko::mpi::communicator comm;
+    int rank;
+};
+
+
+TEST_F(Communicator, CommKnowsItsSize)
+{
+    int size;
+    MPI_Comm_size(MPI_COMM_WORLD, &size);
+
+    EXPECT_EQ(comm.size(), size);
+}
+
+
+TEST_F(Communicator, CommKnowsItsRank)
+{
+    int rank;
+    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+
+    EXPECT_EQ(comm.rank(), rank);
+}
+
+
+TEST_F(Communicator, CommKnowsItsLocalRank)
+{
+    int rank;
+    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+
+    // Expect local rank to be same as rank when on one node
+    EXPECT_EQ(comm.node_local_rank(), rank);
+}
+
+
+TEST_F(Communicator, CommunicatorCanBeCopyConstructed)
+{
+    gko::mpi::communicator copy(comm);
+
+    EXPECT_TRUE(copy == comm);
+}
+
+
+TEST_F(Communicator, CommunicatorCanBeCopyAssigned)
+{
+    gko::mpi::communicator copy = comm;
+
+    EXPECT_TRUE(copy == comm);
+}
+
+
+TEST_F(Communicator, CommunicatorCanBeMoveConstructed)
+{
+    gko::mpi::communicator comm2(MPI_COMM_WORLD);
+    gko::mpi::communicator copy(std::move(comm2));
+
+    EXPECT_TRUE(copy == comm);
+}
+
+
+TEST_F(Communicator, CommunicatorCanBeMoveAssigned)
+{
+    gko::mpi::communicator comm2(MPI_COMM_WORLD);
+    gko::mpi::communicator copy = std::move(comm2);
```

Comment:
Again, I believe this also calls the move constructor, not the move assignment operator. I think you need to separate the initialization and assignment into separate statements.
```suggestion
    gko::mpi::communicator copy(MPI_COMM_NULL);
    copy = std::move(comm2);
```